garden_ai/utils/_meta.py

```diff
@@ -135,15 +136,21 @@ class _USER_PIPELINE_MODULE:
 
     # run the user's code as `__main__._USER_PIPELINE_MODULE` namespace
     local_namespace: dict = {}
+
     exec_getsource(code_str, __main__.__dict__, local_namespace)
-    cls = local_namespace["_USER_PIPELINE_MODULE"]
 
+    cls = local_namespace["_USER_PIPELINE_MODULE"]
     # Now, one of those class attributes is going to be a Pipeline instance
     for name, value in vars(cls).items():
         if isinstance(value, Pipeline):
-            # use whatever identifier the user assigned the pipeline to in their code
+            if not Model.has_been_called:
+                raise PipelineLoadException(
```

Comment:
I think we should change this to just logging a warning instead of raising an exception, since there might be edge cases where users don't use a model in their pipeline at all (like a garden to compare a non-ML approach to an ML approach to the same scientific problem)

Reply:
Fixed, thanks!